### PR TITLE
font_cache: Handle filtering `@font-face` rules in Servo

### PR DESCRIPTION
**Pull request by mrobinson**
Instead of letting Stylo filter `@font-face` rules, handle this
filtering in Servo. It doesn't make sense that Stylo knows about what
fonts Servo supports. This also cleans up a bit the way that this is
handled, giving an entire stylesheet of rules to the font cache to
process instead of letting each layout thread walk the rules. This
brings more of the font-related code into the FontCacheThread itself.

This is the first step toward adding WOFF2 support and fixing various
web font related bugs.

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] There are tests for these changes.

### CODE REVIEW

File: components/gfx/font_cache_thread.rs [hunk 1]
```diff
@@ -624,3 +665,31 @@ impl fmt::Display for LowercaseString {
         self.inner.fmt(f)
     }
 }
+
+fn is_supported_web_font_source(source: &&Source) -> bool {
```
Comment (jdm) [inline components/gfx/font_cache_thread.rs:669]: The double reference here is unusual!

**Review by jdm — APPROVED** [review]
Looking good!

**Comment by mrobinson** [comment]
@jdm Thanks for the review.

The crashes in the test results are due to https://github.com/servo/servo/pull/31597 not landing yet (as it removes a couple `unimplemented()` calls for conic gradients). There are also a few tests that likely need to be rebased.

**Comment by mrobinson** [comment]
So in the end this does change behavior a bit. Now we parse the font language override. No one is using Servo currently to a large extent and not really for the open web, so parsing this value but not using it is fairly low impact. Additionally, we would like to support this value one day.

**Pull request merged**